api/provider/bitbucket.go

```diff
@@ -30,16 +32,52 @@ func (g bitbucketProvider) GetOAuthToken(ctx context.Context, code string) (*oau
 	return g.Exchange(ctx, code)
 }
 
-func (g bitbucketProvider) GetUserEmail(ctx context.Context, tok *oauth2.Token) (string, error) {
+func (g bitbucketProvider) GetUserData(ctx context.Context, tok *oauth2.Token) (*UserProvidedData, error) {
 	u := struct {
 		User struct {
-			Username string `json:"username"`
+			Username  string `json:"username"`
+			FirstName string `json:"first_name"`
+			LastName  string `json:"last_name"`
+			AvatarURL string `json:"avatar"`
 		}
 	}{}
 
 	if err := makeRequest(ctx, tok, g.Config, "https://api.bitbucket.org/1.0/user", &u); err != nil {
+		return nil, err
+	}
+
+	email, err := getUserEmail(ctx, tok, g.Config, fmt.Sprintf("https://api.bitbucket.org/1.0/users/%s/emails", u.User.Username))
+	if err != nil {
+		return nil, err
+	}
+
+	name := strings.TrimSpace(strings.Join([]string{u.User.FirstName, u.User.LastName}, " "))
```

Comment:
In general I don't trust _any_ external data. But I don't actually care that much.